### ISSUE TITLE
Make follow camera avoid obstacles?

### ISSUE THREAD

**Issue by subsoap** [indiesoftby/defold-scene3d#4]
Follow camera does not currently avoid obstacles.

Follow camera could raycast to the player and try to avoid obstacles between it and the player in a smart way.

**Comment by aglitchman** [comment]
Yes, now the included camera following script is basic, and there is room for improvements. 

The camera is an essential part of any game, i.e. camera movement and camera follow can make a massive difference to the player's experience. 

Firstly, let's collect good implementations for inspiration:
- [Tutorial: Orbit Camera - Catlike Coding](https://catlikecoding.com/unity/tutorials/movement/orbit-camera/)
- ["dolly" - Composable camera rigs](https://github.com/h3r2tic/dolly)

**Comment by subsoap** [comment]
Then #5 is related and the issue should be improving the cameras in general.
